lib/chat_web/live/main_live/layout/secret_sharing_form.ex

```diff
@@ -0,0 +1,215 @@
+defmodule ChatWeb.MainLive.Layout.SecretSharingForm do
+  @moduledoc """
+  Handles showing and updating secret sharing as a setting
+  """
+
+  use ChatWeb, :live_component
+
+  import Phoenix.Component
+
+  alias Chat.Card
+  alias Chat.Dialogs.{Dialog, DialogMessaging}
+  alias Chat.Messages.Text
+  alias Chat.User.Registry
+
+  @impl Phoenix.LiveComponent
+  def mount(socket) do
+    users =
+      Registry.all()
+      |> Stream.map(fn {pub_key, %Card{} = user} -> {pub_key, user.name} end)
+      |> Enum.to_list()
+      |> Enum.sort_by(&elem(&1, 1))
+
+    {:ok,
+     socket
+     |> assign(:secret_holders, [])
+     |> assign(:user_options, users)
+     |> assign(:selected_pb_key, nil)
+     |> assign(:selected_kind, nil)
+     |> assign(:threshold, 3)}
```

Comment:
You may use `ok/1` and `noreply/1` helpers in the callback responses.
This is not necessary, but just FYI